File: Day3/twoCityScheduling_Donghee.cpp
class Solution {
public:
    int twoCitySchedCost(vector<vector<int>>& costs) {
        vector<int> diff(); // difference = A - B
        
        priority_queue<pair<int, int> > pq; 
        
        int personId = 0;
        for (vector<int>& pr : costs) {
            pq.push(make_pair(pr[0] - pr[1], personId++));
        }
        
        int count = 0, limit = costs.size() / 2;
        int res = 0;
        
        while (count < limit) {
            // pq.top().second is the personID of a person we should schedule to B rather than A
            res += costs[pq.top().second][1];           
            count++;
            pq.pop();
        }
        
        while (!pq.empty()) {
            res += costs[pq.top().second][0];
            pq.pop();
        }
        
        return res;
    }
};

File: Day3/twoCityScheduling_Donghee.py
class Solution:
    def twoCitySchedCost(self, costs: List[List[int]]) -> int:
        diff = [[0, 0] for x in range(len(costs))] # diff = A - B
        
        for i in range(len(costs)) :
            diff[i] = [costs[i][0] - costs[i][1], i]

        diff.sort(reverse = True) # big to small
        
        count = 0
        limit = len(costs) / 2
        res = 0
        
        while (count < limit) :
            # diff[count][1] is the personID of whom should go to city B
            res += costs[diff[count][1]][1]
            count += 1
            
        while (count < len(costs)) :
            res += costs[diff[count][1]][0]
            count += 1
        
        return res